src/Adapter/Order/AbstractOrderHandler.php

```diff
@@ -243,15 +234,81 @@ protected function updateSpecificPrice(
         $specificPrice->id_product = $product->id;
         $specificPrice->id_product_attribute = $combination ? $combination->id : 0;
         $specificPrice->price = (float) (string) $priceTaxExcluded;
-        $specificPrice->from_quantity = 1;
+        $specificPrice->from_quantity = SpecificPrice::ORDER_DEFAULT_FROM_QUANTITY;
         $specificPrice->reduction = 0;
         $specificPrice->reduction_type = 'amount';
         $specificPrice->reduction_tax = !$priceTaxIncluded->equals($priceTaxExcluded);
-        $specificPrice->from = '0000-00-00 00:00:00';
-        $specificPrice->to = '0000-00-00 00:00:00';
+        $specificPrice->from = SpecificPrice::ORDER_DEFAULT_DATE;
+        $specificPrice->to = SpecificPrice::ORDER_DEFAULT_DATE;
         $specificPrice->add();
     }
 
+    /**
+     * @param Product $product
+     * @param Order $order
+     * @param Combination|null $combination
+     *
+     * @return SpecificPrice|null
+     */
+    protected function getProductSpecificPriceInOrder(
+        Product $product,
+        Order $order,
+        ?Combination $combination
+    ): ?SpecificPrice {
+        // WARNING: DO NOT use SpecificPrice::getSpecificPrice as it filters out fields that are not in database
+        // hence it ignores the customer or cart restriction and results are biased
+        $existingSpecificPriceId = SpecificPrice::exists(
+            $product->id,
+            $combination ? $combination->id : 0,
+            0,
+            0,
+            0,
+            $order->id_currency,
+            $order->id_customer,
+            SpecificPrice::ORDER_DEFAULT_FROM_QUANTITY,
+            SpecificPrice::ORDER_DEFAULT_DATE,
+            SpecificPrice::ORDER_DEFAULT_DATE,
+            false,
+            $order->id_cart
+        );
+
+        if (!empty($existingSpecificPriceId)) {
+            return new SpecificPrice($existingSpecificPriceId);
+        }
+
+        return null;
```

Comment:
```suggestion
        if (empty($existingSpecificPriceId)) {
            return null;
        }

        return new SpecificPrice($existingSpecificPriceId);
```

Fail fast instead, at least in the future if we add a condition for failing it will be easier :)

Reply:
Seriously there is no gain in this one 😅. But just to avoid losing time on this +1month long PR I'll go for it 😉

Reply:
It was just a suggestion this is why it's just a comment and not a "Request change" :sweat_smile: